### PR TITLE
main/lz4: upgrade to 1.8.0

### PR DESCRIPTION
**Pull request by itoffshore**
main/lz4: update to `1.8.0`

### CODE REVIEW

File: main/lz4/APKBUILD [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
 builddir="$srcdir"/$pkgname-$pkgver
 
 build() {
+	return 0
+}
+
+check() {
 	cd "$builddir"
-	make PREFIX="/usr" || return 1
+	# also builds lz4
```
Comment (misery) [inline main/lz4/APKBUILD]: does "abuild build" still works? It should be possible to build without "check"

File: main/lz4/APKBUILD [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
+	make -j4 PREFIX="/usr" test
```
Comment (nmeum) [inline main/lz4/APKBUILD]: Why do you explicitly specify `-j4`? Normally make should pickup `MAKEFLAGS` from `/etc/abuild.conf`

File: main/lz4/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline main/lz4/APKBUILD]: using:

```
build() {
        abuild build
}
```
gives:
```

musl64 [~/aports/main/lz4]$ abuild -r
>>> lz4: Checking sanity of /home/stuart/aports/main/lz4/APKBUILD...
>>> lz4: Analyzing dependencies...
>>> lz4: Installing for build: build-base
(1/1) Installing .makedepends-lz4 (0)
OK: 294 MiB in 145 packages
>>> lz4: Cleaning temporary build dirs...
>>> lz4: Checking sha512sums...
lz4-1.8.0.tar.gz: OK
parallel-tests.patch: OK
>>> lz4: Unpacking /var/cache/distfiles/lz4-1.8.0.tar.gz...
>>> lz4: parallel-tests.patch
patching file README.md
patching file tests/Makefile
>>> ERROR: : Could not find ./APKBUILD (PWD=/home/stuart/aports/main/lz4/src/lz4-1.8.0)
>>> ERROR: lz4: all failed
>>> lz4: Uninstalling dependencies...
(1/1) Purging .makedepends-lz4 (0)
OK: 294 MiB in 144 packages
```

Comment (misery) [inline main/lz4/APKBUILD]: Oh, I did not mean to put it into the APKBUILD. You can call it to build an APKBUILD. Like abuild -r

Comment (nmeum) [inline main/lz4/APKBUILD]: That's not what he meant. You removed the make invocation from build() thereby the software is only compiled as a prerequisites for the test suite in check() and abuild build doesn't build the software anymore.

File: main/lz4/APKBUILD (hunk 2 above)
Comment (itoffshore) [inline main/lz4/APKBUILD]: on my system with `abuild.conf` set to use `MAKEFLAGS` of `-j6` this is too many parallel processes for the unit tests to succeed (see https://github.com/lz4/lz4/issues/385#issuecomment-323869527)

Explicitly using `-j5` & `-j6` also randomly failed sometimes / sometimes not.

Setting `-j4` explicitly ensures the build completes.

Comment (nmeum) [inline main/lz4/APKBUILD]: Wow, that's strange. Could you add a comment with a link to the GitHub issue comment you linked to the APKBUILD?

File: main/lz4/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline main/lz4/APKBUILD]: I've put a `make` in `build()` & a `make test` in `check()`.

(the build was only working with a single `make test` before the parallel build patch)

File: main/lz4/APKBUILD (hunk 2 above)
Comment (itoffshore) [inline main/lz4/APKBUILD]: I've added the comment to the `APKBUILD` for the `lz4` issue & corrected the PR title to `upgrade` from `update`

**Comment by nmeum** [comment]
Ok, thanks for your changes, looking good so far, but I just built this locally and noticed the following by running `checkapk`:

```
--- filelist-lz4-doc-old	2017-08-23 01:50:42.830355432 +0200
+++ filelist-lz4-doc	2017-08-23 01:50:42.833688762 +0200
@@ -2,8 +2,7 @@
 usr/
 usr/share/
 usr/share/man/
-usr/share/man/man1/
-usr/share/man/man1/lz4.1.gz
-usr/share/man/man1/lz4c.1.gz
-usr/share/man/man1/lz4cat.1.gz
-usr/share/man/man1/unlz4.1.gz
+usr/share/man/lz4.1.gz
+usr/share/man/lz4c.1.gz
+usr/share/man/lz4cat.1.gz
+usr/share/man/unlz4.1.gz
```

You probably need to set `MANDIR=/usr/share/man/man1` when invoking `make install` looks like their buildsystem broke with `1.8.0`.

**Comment by itoffshore** [comment]
ok fixed the `man` pages & rebased - will open an issue on `lz4` too.
```
musl64 [~/aports/main/lz4]$ checkapk
>>> No soname differences for lz4.
--- filelist-lz4-dev-old
+++ filelist-lz4-dev
@@ -3,6 +3,7 @@
 usr/include/
 usr/include/lz4.h
 usr/include/lz4frame.h
+usr/include/lz4frame_static.h
 usr/include/lz4hc.h
 usr/lib/
 usr/lib/liblz4.a
>>> No soname differences for lz4-dev.
>>> No soname differences for lz4-doc.
--- filelist-lz4-libs-old
+++ filelist-lz4-libs
@@ -2,4 +2,4 @@
 usr/
 usr/lib/
 usr/lib/liblz4.so.1
-usr/lib/liblz4.so.1.7.5
+usr/lib/liblz4.so.1.8.0
```

**Comment by itoffshore** [comment]
`man` directory issue now fixed in https://github.com/lz4/lz4/pull/390 & applied in `fix-man-dir.patch`

**Pull request closed without merging**